### PR TITLE
Fix incorrect CAPTCHA drag image on scaled Linux display

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/22180

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Linux settings -> Displays -> Scale -> 100%
- Clean profile
- Enable Rewards
- Claim grant
- Solve CAPTCHA
- Close browser
- Linux settings -> Displays -> Scale -> 200%
- Clean profile
- Enable Rewards
- Claim grant
- Solve CAPTCHA

We should also test this on Mac and Windows at various resolutions, although there should be no changes on either of those platforms as this fix is scoped to Linux systems running with scaled displays.

### CODE REVIEW

**Comment by emerick** [comment]
@tmancey mentioned recently that this is no longer a high priority as every 2 hours we fetch new token issuers and when we fetch these we refill tokens.

File: components/brave_rewards/resources/rewards_panel/components/grant_captcha_challenge.tsx [hunk 1]
```diff
@@ -39,6 +39,16 @@ export function GrantCaptchaChallenge (props: Props) {
       x: (rect.width / 2) - (event.clientX - rect.left),
       y: (rect.height / 2) - (event.clientY - rect.top)
     })
+
+    // Work around Linux-specific upstream bug by explicitly setting the drag
+    // image offset when initiating the drag
+    // See https://bugs.chromium.org/p/chromium/issues/detail?id=1297990
+    if (window.navigator.userAgent.includes('Linux') && devicePixelRatio > 1) {
+      event.dataTransfer.setDragImage(
```
Comment (zenparsing) [inline components/brave_rewards/resources/rewards_panel/components/grant_captcha_challenge.tsx:47]: Do you think this is this something that will break the captcha (for this subset of users) when the upstream bug is fixed?

Comment (emerick) [reply components/brave_rewards/resources/rewards_panel/components/grant_captcha_challenge.tsx:47]: I think it's possible, unfortunately, depending on how https://bugs.chromium.org/p/chromium/issues/detail?id=1297990 is ultiumately addressed. I tried to limit the impact as much as possible with the platform and `devicePixelRatio` guards. I feel like any workaround we implement will have to be removed and uplifted once the upstream bug is fixed. Definitely open to other approaches though!

**Review by zenparsing — APPROVED** [review]

**Review by szilardszaloki — APPROVED** [review]

**Comment by emerick** [comment]
CI passed on all platforms but Windows; restarting on Windows.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `PopOS 21.20 x64` using the following build(s):

```
Brave | 1.39.74 Chromium: 101.0.4951.34 (Official Build) nightly (64-bit)
Revision | 8f93af5d759634c5256b65997dedd138ed60992f-refs/branch-heads/4951@{#710}
OS | Linux
```

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.40.7 Chromium: 101.0.4951.41 (Official Build) nightly (64-bit)
-- | --
Revision | 93c720db8323b3ec10d056025ab95c23a31997c9-refs/branch-heads/4951@{#904}
OS | Windows 11 Version 21H2 (Build 22000.613)
```

**`Scaling`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
--------------|----------------|----------------|----------------|-----------------|----------------
scaling100 | grantpanel1 | grantpanel2 | grantpanel3 | settingsPanel1 | settingsPanel2
scaling125 | panelgrant1 | panelgrant2 | panelgrant3 | settingspanel1 | settingsPanel2
scaling150 | panelgrant1 | panelgrant2 | panelgrant3 | settingspanel1 | settingspanel2

**Pull request merged**